akka-remote/src/test/scala/akka/remote/artery/LargeMessagesStreamSpec.scala

```diff
@@ -0,0 +1,157 @@
+        largeRemote.tell(Ping(), senderProbeA.ref)
+        senderProbeA.expectMsg(Pong(0))
+
+        // flag should be cached now
+        largeRemote.asInstanceOf[RemoteActorRef].cachedLargeMessageDestinationFlag should ===(RegularDestination)
+
+      } finally {
+        TestKit.shutdownActorSystem(systemA)
+        TestKit.shutdownActorSystem(systemB)
+      }
+    }
+
+    "pass small regular messages over the large-message stream" in {
+      val systemA = ActorSystem("systemA", config)
+      val systemB = ActorSystem("systemB", config)
+
+      try {
+        val senderProbeA = TestProbe()(systemA)
+        val senderProbeB = TestProbe()(systemB)
+
+        // start actor and make sure it is up and running
+        val large = systemB.actorOf(Props(new EchoSize), "large")
+        large.tell(Ping(), senderProbeB.ref)
+        senderProbeB.expectMsg(Pong(0))
+
+        // communicate with it from the other system
+        val addressB = systemB.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+        val rootB = RootActorPath(addressB)
+        val largeRemote = awaitResolve(systemA.actorSelection(rootB / "user" / "large"))
+        largeRemote.tell(Ping(), senderProbeA.ref)
+        senderProbeA.expectMsg(Pong(0))
+
+        // flag should be cached now
+        largeRemote.asInstanceOf[RemoteActorRef].cachedLargeMessageDestinationFlag should ===(LargeDestination)
+
+      } finally {
+        TestKit.shutdownActorSystem(systemA)
+        TestKit.shutdownActorSystem(systemB)
+      }
+    }
+
+    "allow for normal communication while simultaneously sending large messages" in {
+      val systemA = ActorSystem("systemA", config)
+      val systemB = ActorSystem("systemB", config)
+
+      try {
+
+        val senderProbeB = TestProbe()(systemB)
+
+        // setup two actors, one with the large flag and one regular
+        val large = systemB.actorOf(Props(new EchoSize), "large")
+        large.tell(Ping(), senderProbeB.ref)
+        senderProbeB.expectMsg(Pong(0))
+
+        val regular = systemB.actorOf(Props(new EchoSize), "regular")
+        regular.tell(Ping(), senderProbeB.ref)
+        senderProbeB.expectMsg(Pong(0))
+
+        // both up and running, resolve remote refs
+        val addressB = systemB.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+        val rootB = RootActorPath(addressB)
+        val largeRemote = awaitResolve(systemA.actorSelection(rootB / "user" / "large"))
+        val regularRemote = awaitResolve(systemA.actorSelection(rootB / "user" / "regular"))
+
+        // send a large message, as well as regular one
+        val largeProbe = TestProbe()(systemA)
+        val regularProbe = TestProbe()(systemA)
+
+        val largeBytes = 500000
+        largeRemote.tell(Ping(ByteString.fromArray(Array.ofDim[Byte](largeBytes))), largeProbe.ref)
+        regularRemote.tell(Ping(), regularProbe.ref)
+
+        // this is racy but I don'd know how to test it any other way
```

Comment:
I haven't observed it, just thinking that there is no guarantee that the larger one always completes slower than the regular one.

Which was what I was testing for with the old one by using just one probe, I realize now. So now it just tests that both ping-pongs work, not that the regular one can complete while the large-stream is busy.